webview.h

```diff
@@ -848,24 +848,145 @@ using browser_engine = detail::cocoa_wkwebview_engine;
 //
 
 #define WIN32_LEAN_AND_MEAN
+#include <shellscalingapi.h>
 #include <shlobj.h>
 #include <shlwapi.h>
 #include <stdlib.h>
 #include <windows.h>
-#include <winrt/Windows.Foundation.h>
 
 #include "webview2.h"
 
-#pragma comment(lib, "user32.lib")
-#pragma comment(lib, "Shlwapi.lib")
-#pragma comment(lib, "windowsapp")
+#pragma comment(lib, "ole32.lib")
 #pragma comment(lib, "shell32.lib")
+#pragma comment(lib, "shlwapi.lib")
+#pragma comment(lib, "user32.lib")
 
 namespace webview {
 namespace detail {
 
 using msg_cb_t = std::function<void(const std::string)>;
-using namespace winrt;
+
+std::wstring widen_string(const std::string &narrow_string) {
+  if (narrow_string.empty()) {
+    return std::wstring();
+  }
+  UINT cp = CP_UTF8;
+  DWORD flags = MB_ERR_INVALID_CHARS;
+  auto narrow_string_c = narrow_string.c_str();
+  auto narrow_string_length = static_cast<int>(narrow_string.size());
+  auto required_length = MultiByteToWideChar(cp, flags, narrow_string_c,
+                                             narrow_string_length, nullptr, 0);
+  if (required_length > 0) {
+    std::wstring wide_string(static_cast<std::size_t>(required_length), '\0');
+    if (MultiByteToWideChar(cp, flags, narrow_string_c, narrow_string_length,
+                            wide_string.data(), required_length) > 0) {
+      return wide_string;
+    }
+  }
+  // Failed to convert string from UTF-8 to UTF-16
+  return std::wstring();
+}
+
+std::string narrow_string(const std::wstring &wide_string) {
+  if (wide_string.empty()) {
+    return std::string();
+  }
+  UINT cp = CP_UTF8;
+  DWORD flags = WC_ERR_INVALID_CHARS;
+  auto wide_string_c = wide_string.c_str();
+  auto wide_string_length = static_cast<int>(wide_string.size());
+  auto required_length =
+      WideCharToMultiByte(cp, flags, wide_string_c, wide_string_length, nullptr,
+                          0, nullptr, nullptr);
+  if (required_length > 0) {
+    std::string narrow_string(static_cast<std::size_t>(required_length), '\0');
```

Comment:
we can `::resize` existing string to avoid memory allocations

Reply:
I am not really sure we are on the same page here. Could you please elaborate?

Reply:
Well, I double checked and this is not an issue